File: day_01/day_01.py
with open('./input/input') as file:
    text = file.readlines()

calibrated_numbers = []
for line in text:
    num = ''.join(c for c in line if c.isdigit())
    
    if len(num) == 1:
        num = num * 2
    elif len(num) > 2:
        num = num[0] + num[-1]
    
    calibrated_numbers.append(int(num))

answer = sum(calibrated_numbers)

print("Sum: ", answer)

File: day_01/day_01.cc
#include <iostream>
#include <string>
#include <list>
#include <fstream>

int main(int argc, char* argv[]) {
    if (argc < 2) {
        std::cout << "Usage: " << argv[0] << " <filename>\n";
        return 1;
    }

    std::ifstream input(argv[1]);
    if (!input.is_open()) {
        std::cout << "Failed to open file." << std::endl;
        return 1;
    }

    std::string buffer;
    std::list<std::string> calibration;
    while (std::getline(input, buffer)) {
        std::string digits;
        for (char ch : buffer) {
            if (std::isdigit(ch)) {
                digits.push_back(ch);
            }
        }
        if (!digits.empty()) {
            if (digits.size() == 1) {
                // Duplicate the digit
                digits += digits;
            } else if (digits.size() >= 3) {
                // Keep only the first and last digit
                digits = digits.front() + std::string(1, digits.back());
            }
            // If the size is 2, keep it as is
            calibration.push_back(digits);
        }
    }

    int sum = 0;
    for (const auto& numStr : calibration) {
        sum += std::stoi(numStr);
    }

    std::cout << "Sum: " << sum << std::endl;
    return 0;
}